code/datums/components/extradimensional_storage/extradimensional_prefabs/_prefab_parent.dm

```diff
@@ -0,0 +1,48 @@
+/datum/component/extradimensional_storage/prefab
+	/// The path of the prefab to use.
+	var/prefab_path = null
+	/// The atoms within the prefab that objects entering the prefab may be sent to.
+	var/list/atom/entrances = null
+
+TYPEINFO(/datum/component/extradimensional_storage/prefab)
+	initialization_args = list(
+		ARG_INFO("prefab_path", DATA_INPUT_TYPE, "Prefab path"),
+	)
+
+/datum/component/extradimensional_storage/prefab/Initialize(prefab_path = null)
+	if (!ispath(prefab_path, /datum/mapPrefab/allocated))
+		return COMPONENT_INCOMPATIBLE
+
+	src.prefab_path = prefab_path
+	. = ..()
+
+/datum/component/extradimensional_storage/prefab/UnregisterFromParent()
+	src.entrances = null
+	. = ..()
+
+/datum/component/extradimensional_storage/prefab/set_up_allocated_region()
+	var/datum/mapPrefab/allocated/prefab = global.get_singleton(src.prefab_path)
+	src.region = prefab.load()
+
+	src.entrances = list()
+	for_by_tcl(entrance, /datum/component/extradimensional_prefab_entrance)
+		if (!src.region.turf_in_region(get_turf(entrance.parent)))
+			continue
+
+		src.entrances += entrance.parent
+
+	for_by_tcl(exit, /datum/component/extradimensional_prefab_exit)
+		if (!src.region.turf_in_region(get_turf(exit.parent)))
+			continue
+
+		src.RegisterSignal(exit, COMSIG_EXTRADIMENSIONAL_PREFAB_EXIT, PROC_REF(try_exit))
```

Comment:
This code looks smelly to me. Surely there's a clearer solution than this?

Reply:
I thought that too, but I don't think there is any way of getting a turf's region without iterating through each region and doing a similar check as above, except on the two components?

Reply:
> Would it be faster to just `for_by_tcl` once on `/datum/component` and then do an `istype()` on both entrance and exit respectively?

No, `for_by_tcl` is typecheckless.
It works by iterating over all elements of an entry in the `by_type` list, in this case `by_type[/datum/component/extradimensional_prefab_entrance/exit]`, which contains all current instances of those types. The type lists are set up by the `START/STOP_TRACKING` macros.